### PR TITLE
Day 3

### PR DESCRIPTION
**Pull request by hetrox8**
This project is a mobile web application where you can manage your budget: you have a list of transactions associated with a category, so that you can see how much money you spent and on what.
The main language used include :
1. Ruby
2. Rails
3. Csss

Please use these links to the free icons, when you create a new category :
https://www.nicepng.com/png/detail/168-1681394_food-vegetables-fruits-and-sweets-vector-png-different.png

https://img.freepik.com/free-vector/food-line-icons-collection-with-fruits-drinks-chicken-fish-ice-cream-cake-donut-sausage-pretzel-kitchen_1284-45767.jpg?w=360&t=st=1685873202~exp=1685873802~hmac=2a3c54d95d05b0a84c5abb9586faa80c06afeb2826dadce91bc06e4ca690d1e7

https://media.istockphoto.com/id/1129610282/vector/food-ingredients-and-dishes-icons.jpg?s=1024x1024&w=is&k=20&c=g-VQNeTbTiTkFvx5yZTwL7OywDM5WP5Nijsh9xzMvw0=

## polite notice 
I am really sorry to inform you this i hope you understand. my computer audio isn't working and i am using Ubuntu. i am not able to attach a loom video link . kindly  understand  i'll appreciate thank you

### CODE REVIEW

File: app/controllers/transactions_controller.rb [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+    @transaction = Transaction.new
   end
 
   def create
     @category = Category.find(params[:category_id])
-    @transaction = @category.transactions.build(transaction_params.merge(author_id: current_user.id))
+    @transaction = Transaction.new(transaction_params.merge(author_id: current_user.id))
 
     if @transaction.save
       @transaction.transaction_categories.create(category_id: @category.id)
```
Comment (MahmoudBakr23) [inline app/controllers/transactions_controller.rb:18]: - [ ] Please re-investigate the transactions create method in order to resolve this issue. 🙏  
Action Controller_ Exception caught - Google Chrome 11_23_2023 12_38_33 PM

File: README.md [hunk 2]
```diff
@@ -1,24 +1,143 @@
...
-* Database initialization
+**Budget App** is a Ruby on Rails capstone project which you can create new categories, you can write new transactions and deploy it the categories and finally you can control how much money you spend each category.
 
-* How to run the test suite
+## Video Link 🚀
 
-* Services (job queues, cache servers, search engines, etc.)
 
-* Deployment instructions
+## Render Deploy 🚀
```
Comment (MahmoudBakr23) [inline README.md:34]: - [ ] Please deploy the latest working version of your app and attach the link to the README file as per project requirements. You can use `onrender` to easily deploy a RoR app.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,143 @@
...
-* Configuration
 
-* Database creation
+# 📖 Budget App <a name="about-project"></a>
 
-* Database initialization
+**Budget App** is a Ruby on Rails capstone project which you can create new categories, you can write new transactions and deploy it the categories and finally you can control how much money you spend each category.
 
-* How to run the test suite
+## Video Link 🚀
```
Comment (MahmoudBakr23) [inline README.md]: - [ ] Thank you for communicating your audio issues in the PR summary. However, I'm afraid you'll have to work things around a little bit in order to meet the project requirements. Anyways, here are some steps to check in order to fix your audio problems in Ubuntu OS:

1. Install and Use pavucontrol
pavucontrol is a PulseAudio Volume Control application that provides more advanced control over audio settings. Install it and check the microphone settings.

2. Check PulseAudio Configuration
`sudo nano /etc/pulse/default.pa`

3. Restart PulseAudio
```
pulseaudio -k
pulseaudio --start
```

4. Check for System Updates
```
sudo apt-get update
sudo apt-get upgrade
```

5. Reinstall Audio Drivers
If none of the above steps work, you may want to consider reinstalling audio drivers.

File: README.md [hunk 4]
```diff
@@ -1,24 +1,143 @@
...
+
+I would like to thank Microverse for giving me the opportunity to work on this project. Special thanks Gregoire Vella on Behance who shared his desing.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
+
```
Comment (MahmoudBakr23) [inline README.md]: - [ ] Please fix the MIT link for a professional README file.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @hetrox8 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the controllers' restful methods so far and n+1 query problems
- Great job with authentication so far
- Nice work with the models' validations
- Good work with the database setup, migrations, and the routes
- Well-written unit and integration tests
- Great design
- Descriptive PR
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hetrox8** [comment]
## changes made 
1. fixed transactions
2. Deployed the project
3. Add deployment link in readme file

File: app/controllers/transactions_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/transactions_controller.rb:18]: well resolved! ✔️

File: README.md (hunk 2 above)
Comment (rloterh) [reply README.md:34]: well resolved! ✔️

File: README.md (hunk 3 above)
Comment (rloterh) [reply README.md]: Needs attention

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Suleiman, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🏁 🚦 🏃‍♀️  🌴 🏫 🌳 🏘️

## Highlights 👍  
🎲 Properly added required models, controllers, and corresponding views
🎲 Devise correctly implemented for authentication!
🎲 Schema follow given ERD!
🎲 Correctly deployed app!
🎲 App is running as intended!
🎲 ReadMe looks good!

## Required Changes ♻️
- [ ] Video presentation of the project is missing! Let's endeavor to follow the steps mentioned in the previous review to troubleshoot audio issue or implement an alternative approach to recording a video presentation of your project, which should:
  * be maximum length of the video is 5 minutes
  * contain a clear and objective description about the project in professional English. The topic and purpose of the project are made clear
  * contain a demo of the project, highlighting all the project features.
  * contain a highlight about a piece of the code that you find interesting or is proud of making.

_Kindly check the other comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hetrox8** [comment]
> Hi Suleiman,
> 
> You did extremely great on this project! 👏 👏 There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🏁 🚦 🏃‍♀️ 🌴 🏫 🌳 🏘️
> 
> ## Highlights 👍
> 🎲 Properly added required models, controllers, and corresponding views 🎲 Devise correctly implemented for authentication! 🎲 Schema follow given ERD! 🎲 Correctly deployed app! 🎲 App is running as intended! 🎲 ReadMe looks good!
> 
> ## Required Changes ♻️
> * [ ]  Video presentation of the project is missing! Let's endeavor to follow the steps mentioned in the previous review to troubleshoot audio issue or implement an alternative approach to recording a video presentation of your project, which should:
>   
>   * be maximum length of the video is 5 minutes
>   * contain a clear and objective description about the project in professional English. The topic and purpose of the project are made clear
>   * contain a demo of the project, highlighting all the project features.
>   * contain a highlight about a piece of the code that you find interesting or is proud of making.
> 
> _Kindly check the other comments under the review._
> 
> Cheers and Happy coding!👏 💻 🍷
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hetrox8** [comment]
@rloterh hi bro . i thought i had explained myself about this issue.. i am currently using a desktop computer which technically has no audio/ mic i cant record a thing. in addition to that i am not having any headset or earphones ..

**Comment by hetrox8** [comment]
## changes made 
1. Add loom recording 
https://www.loom.com/share/a1161952049147f4b42e232b8dfdef1e?sid=837dc877-fe8b-40f4-a99b-23f4ad5e59fd

**Review by DammyShittu — APPROVED** [review]
Hi @hetrox8,

## APPROVED 🟢

## Highlights
✔️ Passing tests
✔️ Back buttons on form pages

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**